Intro01/Python/src/main/rapture/App.py

```diff
@@ -0,0 +1,341 @@
+		
+		rawFileData = csvFile.read()
+	
+	# Create blob repo
+	blobRepoUri = "//tutorialBlob"
+	rawCsvUri = blobRepoUri + "/introDataInbound.csv"
+	config = "BLOB {} USING MONGODB {prefix=\"tutorialBlob\"}"
+	metaConfig = "REP {} USING MEMORY {prefix=\"tutorialBlob\"}"
+	if(rapture.doBlob_BlobRepoExists(blobRepoUri)):
+		print "Repo exists, cleaning & remaking"
+		rapture.doBlob_DeleteBlobRepo(blobRepoUri)	
+	rapture.doBlob_CreateBlobRepo(blobRepoUri, config, metaConfig)
+
+	# Encode and Store Blob
+	print "Uploading CSV"
+	encoded_blob = base64.b64encode(str(rawFileData))
+	rapture.doBlob_PutBlob(rawCsvUri, encoded_blob, "text/csv")
+
+	print "CSV successfully uploaded to "+rawCsvUri
+
+def blobToDoc():
+	global rapture
+
+	blobUri = "//tutorialBlob/introDataInbound.csv"
+
+	# Check that blob exists
+	if rapture.doBlob_BlobExists(blobUri):
+
+		# Pull encoded blob out of rapture
+	 	encoded_blob = rapture.doBlob_GetBlob(blobUri)['content']
+		# Decode blob using base64 python library
+		blob_string = base64.b64decode(str(encoded_blob))
+		# Add list identifiers to string so that we can convert to python list
+		blob_string = "[[\"" + blob_string.replace(',', "\",\"").replace('\n', "\"],[\"") + "\"]]"
+		# Convert string to python list
+		rows = ast.literal_eval(blob_string)
+
+		# Create doc repo
+		docRepoUri = "//tutorialDoc"
+		docUri = docRepoUri + "/introDataTranslated"
+		config = "REP {} USING MONGODB {prefix=\"tutorialDoc\"}"
+		if(rapture.doDoc_DocRepoExists(docRepoUri)):
+			rapture.doDoc_DeleteDocRepo(docRepoUri)	
+		rapture.doDoc_CreateDocRepo(docRepoUri, config)
+
+		#CREATE ONE FLAT JSON
+		#CREATING INITIAL STRUCTURE
+		keys = rows[0]
+		rowData = {}
+		docIndex = 0
+		blobData = list()
+		for row in rows[1:]:
+			rowData = {keys[0]:row[0],keys[1]:row[1],keys[2]:row[2],keys[3]:row[3],keys[4]:row[4],keys[5]:row[5],keys[6]:row[6]}
+			blobData.append(rowData)
+		#This is the top level of json/dict that we will be inserting
+		Order ={}
+		for entry in blobData:
+			Order['series_type'] = entry['series_type']
+			Order['frequency'] = entry['frequency']
+			Order['index_id'] = {}
+			Order['provider'] = entry['provider']
+		index_id = list()
+		#Add index_id's to the master json/dict
+		for x in blobData:
+			index_id.append(x['index_id'])
+		index_idSet = set(index_id)
+		index_idDict = {}
+		for x in index_idSet:
+			index_idDict.update({x:{}})
+		Order['index_id'] = index_idDict
+		priceTypeList = list()
+		for x in blobData:
+			priceTypeList.append(x['price_type'])
+		priceTypeSet = set(priceTypeList)
+		priceTypeDict = {}
+		for x in priceTypeSet:
+			priceTypeDict.update({x:{}})
+		indxID = Order['index_id']
+		indx_idUpdate = {}
+		#Add price_types to the master json/dict
+		for x in indxID:
+			indx_idUpdate.update({x:priceTypeDict})
+		Order['index_id'] = indx_idUpdate
+		
+		#CREATE DICTIONARYS OF DATES/PRICES BY INDEX_ID/PRICE_TYPE
+		USG = {}
+		EMN = {}
+		LAK = {}
+		AUD = {}

```

Comment:
Hmm. Can't we do a generic version that doesn't make assumptions about the data values? 
It's one thing to hard code column names, but I'm uncomfortable about making it so specific.